src/core/dyad_core.h

```diff
@@ -0,0 +1,192 @@
+#ifndef __DYAD_CORE_CLIENT_H__
+#define __DYAD_CORE_CLIENT_H__
+
+// Includes <flux/core.h>
+#include "dyad_flux_log.h"
+#include "dyad_err.h"
+
+#ifdef __cplusplus
+#include <cstdlib>
+#include <cstdio>
+#else
+#include <stdbool.h>
+#include <stdlib.h>
+#include <stdio.h>
+#endif
+
+/*****************************************************************************
+ *                                                                           *
+ *                          DYAD Macro Definitions                           *
+ *                                                                           *
+ *****************************************************************************/
+
+#define DYAD_PATH_PROD_ENV "DYAD_PATH_PRODUCER"
+#define DYAD_PATH_CONS_ENV "DYAD_PATH_CONSUMER"
+#define DYAD_CHECK_ENV     "DYAD_SYNC_HEALTH"
+// Now defined in src/utils/utils.h
+//#define DYAD_PATH_DELIM    "/"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+/**
+ * @struct dyad_ctx
+ */
+struct dyad_ctx {
+    flux_t *h; // the Flux handle for DYAD
+    bool debug; // if true, perform debug logging
+    bool check; // if true, perform some check logging
```

Comment:
Not sure what the purpose of check flag is for

Reply:
I'm not positive either. It's leftover from the definition of the context in `wrapper.h` (in the `main` branch). If it's not necessary, I'll get rid of it.